tests/papi-client.spec.ts

```diff
@@ -0,0 +1,231 @@
+            expectBatchStart();
+            mock.get("http://obm.org/my-domain/batches/123/", (req) => {
+                expect(req.headers.authorization).to.equal("Basic YWRtaW46cHdk");
+                return {body: {}};
+            });
+
+            papiClient.startBatch().then(papiClient.waitForBatchSuccess.bind(papiClient));
+        });
+
+        it("should reject the promise if the request is in error", (done) => {
+            expectBatchStart();
+            mock.get("http://obm.org/my-domain/batches/123/", () => {
+                return {status: 500};
+            });
+
+            papiClient.startBatch()
+                .then(papiClient.waitForBatchSuccess.bind(papiClient))
+                .catch((err) => {
+                    expect(err).to.deep.equal({status: 500});
+                    done();
+                });
+        });
+
+        it("should reject the promise when the status is ERROR", (done) => {
+            expectBatchStart();
+            mock.get("http://obm.org/my-domain/batches/123/", () => {
+                return {body: {status: "ERROR", operationDone: 2, operationCount: 5}};
+            });
+
+            papiClient.startBatch()
+                .then(papiClient.waitForBatchSuccess.bind(papiClient))
+                .catch((err) => {
+                    expect(err).to.equal("ERROR: 2/5");
+                    done();
+                });
+        });
+
+        it("should resolve the promise when the status is SUCCESS", (done) => {
+            expectBatchStart();
+            mock.get("http://obm.org/my-domain/batches/123/", () => {
+                return {body: {status: "SUCCESS", operationCount: 5, operationDone: 5}};
+            });
+
+            papiClient.startBatch()
+                .then(papiClient.waitForBatchSuccess.bind(papiClient))
+                .then(
+                    (res) => {
+                        expect(res).to.equal("SUCCESS: 5/5");
+                        expect(papiClient.currentBatchId).to.be.undefined;
+                        done();
+                    },
+                    done
+                );
+        });
+
+        it("should retry as long as the batch is not done", (done) => {
+            expectBatchStart();
+
+            let queryCount = 0;
+            mock.get("http://obm.org/my-domain/batches/123/", (): any => {
+                queryCount++;
+                if (queryCount === 3) {
+                    return {body: {status: "SUCCESS", operationCount: 5, operationDone: 5}};
+                }
+                return {body: {status: "RUNNING"}};
+            });
+
+            papiClient.startBatch()
+                .then(papiClient.waitForBatchSuccess.bind(papiClient, 10))
+                .then(
+                    () => {
+                        expect(queryCount).to.equal(3);
+                        done();
+                    },
+                    done
+                );
+        });
+
```

Comment:
do you handle the case where the batch response status is marked SUCCESS, but there are inconsistencies between operationCount and operationDone (for instance!) ? Servers can sometimes have awkward behaviors/response :)